### ISSUE TITLE
API: introduce /<bookname>/wp-json/pressbooks/v2/metadata endpoint

### ISSUE THREAD

**Issue by greatislander** [pressbooks/pressbooks#771]
### Prerequisites

* [x] Can you reproduce the behavior you wish to address with [all other plugins deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [x] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/) and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?
* [x] Are you experiencing this issue with one of our built-in root or book themes?
* [x] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fpressbooks) to see if your bug or enhancement is already listed?

### Description

We need an endpoint on each book and each section which returns the full metadata for the book or section. Requirements:

* [x] Returns book information mapped to http://schema.org/Book properties
* [ ] Returns section information mapped to http://bib.schema.org/Chapter properties

### Steps to Reproduce

Not applicable.

### System Information

Pressbooks: 4.0.0-dev
WordPress: 4.7.5
PHP: 7.1

**Comment by connerbw** [comment]
Maybe: https://github.com/readium/webpub-manifest ?

Requirements may change.

**Comment by greatislander** [comment]
Mapping of [Schema.org properties](http://schema.org/Book) to book information (props @bdolor): https://github.com/pressbooks/pressbooks/blob/71f400c50037170b69044e5ee5857e18b16d7902/inc/class-metadata.php#L143-L157

```
$micro_mapping = [
	'about' => 'pb_bisac_subject',
	'alternativeHeadline' => 'pb_subtitle',
	'author' => 'pb_author',
	'contributor' => 'pb_contributing_authors',
	'copyrightHolder' => 'pb_copyright_holder',
	'copyrightYear' => 'pb_copyright_year',
	'datePublished' => 'pb_publication_date',
	'description' => 'pb_about_50',
	'editor' => 'pb_editor',
	'image' => 'pb_cover_image',
	'inLanguage' => 'pb_language',
	'keywords' => 'pb_keywords_tags',
	'publisher' => 'pb_publisher',
];
```

**Comment by greatislander** [comment]
http://bib.schema.org/isBasedOn

☝️ This could be good for indicating canonical version of a cloned book.

**Comment by greatislander** [comment]
@connerbw My feeling would be that we use the metadata endpoint to generate that component of the webpub manifest: https://github.com/readium/webpub-manifest/blob/master/examples/MobyDick/manifest.json#L4-L11

**Comment by hughmcguire** [comment]
OK, I would propose:

```
$micro_mapping = [
	'about' => 'pb_bisac_subject',
	'alternativeHeadline' => 'pb_subtitle',
	'author' => 'pb_author',
	'contributor' => 'pb_contributing_authors',
	'copyrightHolder' => 'pb_copyright_holder',
	'copyrightYear' => 'pb_copyright_year',
	'datePublished' => 'pb_publication_date',
	'description' => 'pb_about_50',
	'editor' => 'pb_editor',
	'image' => 'pb_cover_image',
	'inLanguage' => 'pb_language',
	'keywords' => 'pb_keywords_tags',
	'publisher' => 'pb_publisher',
];

#### things we might want to add support for (more details to come).
numberOfPages 
accessibilityAPI
accessibilitySummary
audience
comment 
discussionUrl
editor
educationalAlignment	
educationalUse
genre
inLanguage
isBasedOn
license
publishingPrinciples
review
sourceOrganization
thumbnailUrl
translator
version

#### Properties from Thing
description
identifier
image
url

Todo: `Person`

**Comment by greatislander** [comment]
See also: https://raw.githubusercontent.com/readium/webpub-manifest/master/contexts/default/context.jsonld

**Comment by hughmcguire** [comment]
see also: https://www.w3.org/TR/dpub-metadata/

**Comment by bdolor** [comment]
If I may...

LRMI Additions to Schema.org (reference http://dublincore.org/dcx/lrmi-terms/1.1/)
```
educationalAlignment (added above already by Hugh)
educationalUse (added above already by Hugh)
timeRequired
typicalAgeRange
interactivityType
learningResourceType
isBasedOnUrl
```

all items above conform to Schema.org Book (reference: http://schema.org/Book) 

Pressbooks-Textbook: https://github.com/BCcampus/pressbooks-textbook/blob/dev/themes-book/opentextbook/functions.php#L66

**Comment by greatislander** [comment]
Looks good, @bdolor!

**Comment by hughmcguire** [comment]
@greatislander see above I edited my laundry list.

**Comment by hughmcguire** [comment]
A note about metadata for us as we think about UX that sits atop this: authors & regular people hate inputting metadata, makes them confused & frustrated. So, the core things should be front & centre, and additional attributes need to be ... obscured somehow, so as not to overwhelm.

**Comment by greatislander** [comment]
@hughmcguire ya

**Comment by hughmcguire** [comment]
More links: 
BISG (Book industry study group) will be releasing shortly a Schema.org >> ONIX map, which means if we do this right we get ONIX output for "free" ... : 
http://bisg.org/page/SchemaorgWG/Schema.org-Working-Group-.htm

**Comment by hughmcguire** [comment]
Collecting various links: 
https://github.com/readium/webpub-manifest
https://raw.githubusercontent.com/readium/webpub-manifest/master/contexts/default/context.jsonld
http://bib.schema.org/isBased
https://www.w3.org/TR/dpub-metadata/
https://github.com/w3c/dpub-metadataOn
http://dublincore.org/dcx/lrmi-terms/1.1/
http://bisg.org/page/SchemaorgWG/Schema.org-Working-Group-.htm

**Comment by greatislander** [comment]
https://idpf.github.io/epub-guides/schema-org-integration/

**Comment by greatislander** [comment]
Implemented (at book level) in #798:

- about
- alternateName
- alternativeHeadline
- author
  - name
  - alternateName
- contributor
  - name
- copyrightHolder
  - name
- copyrightYear
- datePublished
- description
- editor
  - name
- image
- inLanguage
- keywords
- name
- publisher
  - name
  - address
    - addressLocality
- translator
  - name
